src/deepsparse/transformers/pipelines/embedding_extraction.py

```diff
@@ -0,0 +1,375 @@
+        self._model_size = model_size
+        self._extraction_strategy = extraction_strategy
+        self._return_numpy = return_numpy
+
+        if kwargs.get("batch_size", None) != 1:
+            raise ValueError("EmbeddingExtractionPipeline only supports batch size 1")
+
+        if self._extraction_strategy not in ExtractionStrategy.to_list():
+            raise ValueError(
+                f"Unsupported extraction_strategy {self._extraction_strategy}"
+            )
+
+        super().__init__(**kwargs)
+
+    @property
+    def input_schema(self) -> Type[BaseModel]:
+        """
+        :return: pydantic model class that inputs to this pipeline must comply to
+        """
+        return EmbeddingExtractionInput
+
+    @property
+    def output_schema(self) -> Type[BaseModel]:
+        """
+        :return: pydantic model class that outputs of this pipeline must comply to
+        """
+        return EmbeddingExtractionOutput
+
+    def setup_onnx_file_path(self) -> str:
+        """
+        Performs setup done in pipeline parent class as well as truncating the
+        model to an intermediate layer for embedding extraction
+
+        :return: file path to the processed ONNX file for the engine to compile
+        """
+        onnx_path = super().setup_onnx_file_path()
+
+        if self._emb_extraction_layer is not None:
+            (
+                onnx_path,
+                self.onnx_output_names,
+                self._temp_model_directory,
+            ) = truncate_transformer_onnx_model(
+                onnx_path,
+                emb_extraction_layer=self._emb_extraction_layer,
+                hidden_layer_size=self._model_size,
+            )
+        else:
+            _LOGGER.info("Skipping model truncation")
+
+        return onnx_path
+
+    def parse_inputs(self, *args, **kwargs) -> BaseModel:
+        """
+        :param args: ordered arguments to pipeline, either a input_schema object,
+            a string text, or a list of inputs
+        :param kwargs: keyword arguments to pipeline
+        :return: pipeline arguments parsed into the given `input_schema`
+            schema if necessary. If an instance of the `input_schema` is provided
+            it will be returned
+        """
+        if args and kwargs:
+            raise ValueError(
+                f"{self.__class__} only support args OR kwargs. Found "
+                f" {len(args)} args and {len(kwargs)} kwargs"
+            )
+
+        if args:
+            if isinstance(args, str):
+                return self.input_schema(inputs=[args[0]])
+
+            if len(args) == 1:
+                if isinstance(args[0], self.input_schema):
+                    return args[0]
+                else:
+                    return self.input_schema(inputs=args[0])
+            else:
+                return self.input_schema(inputs=args)
+        else:
+            return self.input_schema(**kwargs)
+
```

Comment:
Not a blocking comment, but recommend flipping the if elses to make it more flat & easier to scan (the less indentation the better!). Something like:

```python
if not args:
    return self.input_schema(**kwargs)
if isinstance(args, str):
    return self.input_schema(inputs=[args[0]])
if len(args) > 1 or len(args) == 0:
    return self.input_schema(inputs=args)
if isinstance(args[0], self.input_schema)
    return args[0]
return self.input_schema(inputs=args[0])
```

This format also makes it easier to add conditions in the future (instead of having to add more indentation levels)